packages/commercetools/api-client/src/types/Api.ts

```diff
@@ -3,14 +3,12 @@ import { FetchResult } from 'apollo-link';
 import { Cart, Order, ShippingMethod, CustomerSignInResult, Customer } from './GraphQL';
 import { Token, CustomerCredentials } from './setup';
 
-export interface CustomQuery<T> {
+export type CustomQueryFn = () => {
   query: any;
-  variables: T;
-}
-
-export type CustomQueryFn<T = any> = (query?: any, variables?: T) => CustomQuery<T>;
+  variables?: Record<string, any>;
+};
 
-export const getCustomQuery = <T = any>(customQueryFn: CustomQueryFn<T>, defaultQuery) => {
+export const getCustomQuery = (customQueryFn: CustomQueryFn, defaultQuery) => {
```

Comment:
I know it's @lukasborawski stuff but why the hell we have a function in `types` dir? @lukasborawski can you fix that in some other PR? :)